### PR TITLE
Don't show offset calibration controls when "relax" mod is active

### LINKED ISSUE #22892: Previous play calibration menu showing up with 'Relax' mod
### Type

Game behaviour

### Bug description

When using the Relax mod, if you restart the level the previous play calibration information shows up. This is unexpected, because with Relax you are only moving the cursor and there is no timing that needs to be calibrated.

### Screenshots or videos

https://user-images.githubusercontent.com/44739965/226073918-7f9f8710-90a2-4d60-b43d-fef5f21f6ea9.mp4

### Version

2023.305.0-lazer

### Logs

[updater.log](https://github.com/ppy/osu/files/11007090/updater.log)
[database.log](https://github.com/ppy/osu/files/11007091/database.log)
[input.log](https://github.com/ppy/osu/files/11007092/input.log)
[network.log](https://github.com/ppy/osu/files/11007093/network.log)
[performance.log](https://github.com/ppy/osu/files/11007094/performance.log)
[runtime.log](https://github.com/ppy/osu/files/11007095/runtime.log)

### PR DESCRIPTION
**Pull request by peppy**
Closes https://github.com/ppy/osu/issues/22892.

### CODE REVIEW

File: osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs [hunk 1]
```diff
@@ -183,7 +184,7 @@ private void scoreChanged(ValueChangedEvent<ScoreInfo?> score)
             if (score.NewValue == null)
                 return;
 
-            if (score.NewValue.Mods.Any(m => !m.UserPlayable))
+            if (score.NewValue.Mods.Any(m => !m.UserPlayable || m is ModRelax))
```
Comment (Joehuu) [inline osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs:187]: It should still show with taiko relax (not yet implemented though, see https://github.com/ppy/osu/issues/7914).

Comment (peppy) [reply osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs:187]: I'm not sure taiko relax should return under the "relax" name yet..

Comment (peppy) [reply osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs:187]: Here's an alternative to scope this change down further if preferred:

```diff
diff --git a/osu.Game.Rulesets.Osu/Mods/OsuModRelax.cs b/osu.Game.Rulesets.Osu/Mods/OsuModRelax.cs
index 32ffb545e0..aaa7c70a8d 100644
--- a/osu.Game.Rulesets.Osu/Mods/OsuModRelax.cs
+++ b/osu.Game.Rulesets.Osu/Mods/OsuModRelax.cs
@@ -18,7 +18,7 @@
 
 namespace osu.Game.Rulesets.Osu.Mods
 {
-    public class OsuModRelax : ModRelax, IUpdatableByPlayfield, IApplicableToDrawableRuleset<OsuHitObject>, IApplicableToPlayer
+    public class OsuModRelax : ModRelax, IUpdatableByPlayfield, IApplicableToDrawableRuleset<OsuHitObject>, IApplicableToPlayer, IHasNoTimedInputs
     {
         public override LocalisableString Description => @"You don't need to click. Give your clicking/tapping fingers a break from the heat of things.";
 
diff --git a/osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs b/osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs
index 8d5459809e..b542707185 100644
--- a/osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs
+++ b/osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs
@@ -184,7 +184,7 @@ private void scoreChanged(ValueChangedEvent<ScoreInfo?> score)
             if (score.NewValue == null)
                 return;
 
-            if (score.NewValue.Mods.Any(m => !m.UserPlayable || m is ModRelax))
+            if (score.NewValue.Mods.Any(m => !m.UserPlayable || m is IHasNoTimedInputs))
                 return;
 
             var hitEvents = score.NewValue.HitEvents;

```

Comment (bdach) [reply osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs:187]: I think applying above as proposed is fine.

**Review by bdach — DISMISSED** [review]

**Review by bdach — APPROVED** [review]

**Pull request merged**